### PR TITLE
feat: get and delete sales tracker

### PR DESCRIPTION
**Pull request by hoangharry**
**What does this PR do?**

-   [x] Get nft sales tracker by guild id
-  [x] Delete nft sales tracker by guild id and contract address

**Media**
GET endpoint
Screen Shot 2022-08-26 at 11 37 44

DELETE endpoint
Screen Shot 2022-08-26 at 11 38 16

### CODE REVIEW

File: pkg/entities/nfts.go [hunk 1]
```diff
@@ -654,6 +655,41 @@ func (e *Entity) GetAllNFTSalesTracker() ([]response.NFTSalesTrackerResponse, er
 	return resp, nil
 }
 
+func (e *Entity) DeleteNFTSalesTracker(guildID, contractAddress string) error {
+	salesTracker, err := e.repo.NFTSalesTracker.GetNFTSalesTrackerByContractAndGuildID(guildID, contractAddress)
```
Comment (catngh) [inline pkg/entities/nfts.go:659]: can we delete and err check instead of querying first?

Comment (trkhoi) [inline pkg/entities/nfts.go:659]: can we use DeleteNFTSalesTrackerByAddress

File: pkg/routes/v1.go [hunk 2]
```diff
@@ -203,7 +203,9 @@ func NewRoutes(r *gin.Engine, h *handler.Handler, cfg config.Config) {
 		nftsGroup.GET("/supported-chains", h.GetSupportedChains)
 		nftsGroup.GET("/trading-volume", h.GetNFTTradingVolume)
 		nftsGroup.POST("/sales-tracker", h.CreateNFTSalesTracker)
+		nftsGroup.DELETE("/sales-tracker", h.DeleteNFTSalesTracker)
 		nftsGroup.GET("/sales-tracker", h.GetAllNFTSalesTracker)
+		nftsGroup.GET("/sales-tracker/:guildID", h.GetNFTSaleSTrackerByGuildID)
```
Comment (trkhoi) [inline pkg/routes/v1.go:208]: put guildID in query params to consist with delete

File: pkg/entities/nfts.go [hunk 3]
```diff
@@ -654,6 +655,41 @@ func (e *Entity) GetAllNFTSalesTracker() ([]response.NFTSalesTrackerResponse, er
...
+	if err != nil {
+		e.log.Fields(logger.Fields{"guildID": guildID}).Error(err, "[entity.GetNFTSaleSTrackerByGuildID] failed to get nft sales trackers")
+		return nil, err
+	}
+	if len(data) == 0 {
+		return nil, fmt.Errorf("Guild has no sales trackers")
+	}
+
+	collection := []response.NFTSalesTracker{}
+	for _, v := range data {
```
Comment (trkhoi) [inline pkg/entities/nfts.go:678]: i think can preload nft_sales_tracker

**Review by trkhoi — CHANGES_REQUESTED** [review]

File: pkg/entities/nfts.go (hunk 1 above)
Comment (hoangharry) [reply pkg/entities/nfts.go:659]: query first to get ID to delete because the request doesn't include ID

Comment (hoangharry) [reply pkg/entities/nfts.go:659]: query first to get ID to delete because the request doesn't include ID

File: pkg/entities/nfts.go (hunk 3 above)
Comment (hoangharry) [reply pkg/entities/nfts.go:678]: use preload to query on database but it repeats guildID and channelID several times in collection so i use this loop to remove the unnecessary information

Comment (trkhoi) [reply pkg/entities/nfts.go:678]: unnecessary information will not be used by bot, i think use preload will save time here

File: pkg/entities/nfts.go (hunk 1 above)
Comment (trkhoi) [reply pkg/entities/nfts.go:659]: can query by address instead of id ? will use 1 step only

**Review by trkhoi — APPROVED** [review]

**Review by tuanddd — APPROVED** [review]

**Pull request merged**